doc/connectivity/index.rst

```diff
@@ -9,3 +9,4 @@ Connectivity
    bluetooth/index.rst
    networking/index.rst
    lora_lorawan/index.rst
+   usb_c/index.rst
```

Comment:
does usb-c really belong under connectivity?

Reply:
From my perspective, yes. It's not wireless, but it's definitely connectivity. I think we should move the general USB doc to connectivity in fact. I will send a PR after this one is merged.

Reply:
connectivity here was meant to accomodate comm technologies, wired and wireless. For the sake of this PR going in, we can revisit this later and discusss what falls under connectivity somewhere else.